src/App.tsx

```diff
@@ -1,39 +1,71 @@
 import React from 'react';
 import './App.scss';
 
+import { Clock } from './Clock';
+
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+interface State {
+  clockName: string,
+  hasClock: boolean,
+}
+
+export class App extends React.Component<{}, State> {
+  state = {
+    clockName: getRandomName(),
+    hasClock: true,
+  };
+
+  handleChangeClockName = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleClockTurnOff);
+    document.addEventListener('click', this.handleClockTurnOn);
+    this.handleChangeClockName = window
+      .setInterval(this.handleClockName, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(_: {}, prevState: State) {
+    if (this.state.clockName !== prevState.clockName) {
+      window.console.log(
+        `Renamed from ${prevState.clockName} to ${this.state.clockName}`,
+      );
+    }
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleClockTurnOff);
+    document.removeEventListener('click', this.handleClockTurnOn);
+    window.clearInterval(this.handleChangeClockName);
```

Comment:
window.clearInterval(this.handleChangeClockName); - it wouldn't work here because App don't close the window or update. you should replace this clearance to componentDidUpdate because of changing the state
<img width="305" alt="image" src="https://user-images.githubusercontent.com/106744410/183125139-06fbd5ee-7c3a-4e0b-8d9f-bccee05ed260.png">

Reply:
Thank you! Really good point!